### DISCUSSION TITLE
How do you get the sub-schema of a schema?

### DISCUSSION THREAD

**Discussion by matthew-dean** [open-circle/valibot#653] [Q&A]
Say I have this:

```ts
const Schema = object({
  id: string()
  // other stuff
})
```

Now I want to do this:
```ts
const NewSchema = object({
  id: optional(Schema.id) // ??
})
```
I can't find anything in the documentation about this. _(Note, this is obviously over-simplified, and I know I could do `optional(string())` but the actual sub-schema will be much more complicated.)_

**Comment by fabian-hiller** [comment]
Each schema is fully accessible. You can access the entries of the object schema via `.entries`:

```ts
const NewSchema = object({
  id: optional(Schema.entries.id)
});
```

**Reply by matthew-dean** [reply]
@fabian-hiller Thank you!!

**Reply by matthew-dean** [reply]
@fabian-hiller is `entries`  called out in the documentation somewhere and I just missed it?

**Reply by fabian-hiller** [reply]
Yes, if you click through the API reference, but the most types are not updated to v0.31.0. I will do this in the next few weeks. https://valibot.dev/api/ObjectSchema/
